libs/api/ebpf_api.cpp

```diff
@@ -1475,6 +1481,169 @@ _get_next_map_to_create(std::vector<ebpf_map_t*>& maps)
     return nullptr;
 }
 
+ebpf_result_t
+ebpf_object_open(
+    _In_z_ const char* path,
+    _In_opt_z_ const char* object_name,
+    _In_opt_ const ebpf_program_type_t* program_type,
+    _In_opt_ const ebpf_attach_type_t* attach_type,
+    _Outptr_ struct bpf_object** object,
+    _Outptr_result_maybenull_z_ const char** error_message) noexcept
+{
+    *error_message = nullptr;
+
+    ebpf_object_t* new_object = new (std::nothrow) ebpf_object_t();
+    if (new_object == nullptr) {
+        return EBPF_NO_MEMORY;
+    }
+
+    ebpf_result_t result =
+        _initialize_ebpf_object_from_elf(path, object_name, program_type, attach_type, *new_object, error_message);
+    if (result != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    *object = new_object;
+    _ebpf_objects.emplace_back(*object);
```

Comment:
This can throw. We have new (std::nothrow) above, but we make a call that can throw here? Why?

Reply:
Doesn't appear to be unique to this PR. File #583 to track this. I recommend that we handle this in a systemic way and have very top-level API have a try/except to make sure we don't leak C++ exceptions across a C ABI.

Reply:
agree

Reply:
@Alan-Jowett does that mean you want this instance fixed in this PR or are you still reviewing?

Reply:
I think we should do it systematically as part of another PR.